crates/ruff/src/checkers/ast.rs

```diff
@@ -4364,14 +4363,131 @@ impl<'a> Checker<'a> {
         for scope_index in self.scope_stack.iter().rev() {
             let scope = &self.scopes[*scope_index];
 
-            if matches!(scope.kind, ScopeKind::Class(_)) {
+            if let ScopeKind::Class(def) = &scope.kind {
                 if id == "__class__" {
                     return;
                 } else if !first_iter && !in_generator {
                     continue;
                 }
+                for base in def.bases.iter() {
+                    let mut base_class_name = None;
+                    match &base.node {
+                        ExprKind::Attribute {
+                            value,
+                            attr: class_name,
+                            ..
+                        } => {
+                            if let ExprKind::Name {
+                                id: module_name, ..
+                            } = &value.node
+                            {
+                                base_class_name = Some(format!("{module_name}.{class_name}"));
+                            }
+                        }
+                        ExprKind::Name { id: bcn, .. } => {
+                            base_class_name = Some(bcn.to_string());
+                        }
```

Comment:
Instead of checking based on strings here, it'd be preferable to instead use `self.resolve_call_path`, which will give you back a `CallPath` object, and automatically handles aliased imports etc. The `should_ignore_definition` method in `crates/ruff/src/rules/pydocstyle/helpers.rs` is one example of what that might look like.